Add v0.18.2 changelog (#7083)
aws/eks-anywhere: docs/content/en/docs/whatsnew/changelog.md

* When upgrading to a new minor version, a new OS image must be created using the new image-builder cli.
{{% /alert %}}

## [v0.18.2](https://github.com/aws/eks-anywhere/releases/tag/v0.18.2)

### Fixed

> * RHEL: Fixes `no_proxy` inputs to be properly parsed, when both Red Hat Satellite and Proxy is used in image-builder. ([#2664](https://github.com/aws/eks-anywhere-build-tooling/pull/2664))

Reviewer: ```suggestion
* Image Builder: Correctly parse `no_proxy` inputs when both Red Hat Satellite and Proxy is used in image-builder. ([#2664](https://github.com/aws/eks-anywhere-build-tooling/pull/2664))
```

> * For vSphere provider, pass the full template path when validating template tags instead of just the name during create/upgrade cluster. ([#6437](https://github.com/aws/eks-anywhere/pull/6437))

Reviewer: ```suggestion
* vSphere: Fix template tag validation by specifying the full template path ([#6437](https://github.com/aws/eks-anywhere/pull/6437))
```

This describes a how. What did this fix? Use that as the summarizing statement.

> * For Bare Metal provider, Skip `kube-vip` deployment when `TinkerbellDatacenterConfig.skipLoadBalancerDeployment` is set to true. ([#6990](https://github.com/aws/eks-anywhere/pull/6990))

Reviewer: ```suggestion
* Bare Metal: Skip `kube-vip` deployment when `TinkerbellDatacenterConfig.skipLoadBalancerDeployment` is set to true. ([#6990](https://github.com/aws/eks-anywhere/pull/6990))
```

### Other

> * Patch incorrect conversion between uint64 and int64 ([#7048](https://github.com/aws/eks-anywhere/pull/7048))
> * Fix incorrect regex for matching curated package registry URL ([#7049](https://github.com/aws/eks-anywhere/pull/7049))
> * Patch malicious tarballs directory traversal vulnerability ([#7057](https://github.com/aws/eks-anywhere/pull/7057))

Reviewer: ```suggestion
* Security: Patch incorrect conversion between uint64 and int64 ([#7048](https://github.com/aws/eks-anywhere/pull/7048))
* Security: Fix incorrect regex for matching curated package registry URL ([#7049](https://github.com/aws/eks-anywhere/pull/7049))
* Security: Patch malicious tarballs directory traversal vulnerability ([#7057](https://github.com/aws/eks-anywhere/pull/7057))
```
